clients/importapi/importapi.go

```diff
@@ -0,0 +1,246 @@
+
+	resp, err := c.client.Get(ctx, c.url+"/healthcheck")
+	if err != nil {
+		return service, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return service, NewAPIResponse(resp, "/healthcheck")
+	}
+
+	return service, nil
+}
+
+// ImportJob comes from the Import API and links an import job to its (other) instances
+type ImportJob struct {
+	JobID string  `json:"id"`
+	Links LinkMap `json:"links,ignoreempty"`
+}
+
+// LinkMap is an array of instance links associated with am import job
+type LinkMap struct {
+	Instances []InstanceLink `json:"instances"`
+}
+
+// InstanceLink identifies an (instance or import-job) by id and url (from Import API)
+type InstanceLink struct {
+	ID   string `json:"id"`
+	Link string `json:"href"`
+}
+
+// GetImportJob asks the Import API for the details for an Import job
+func (api *Client) GetImportJob(ctx context.Context, importJobID string) (ImportJob, bool, error) {
+	var importJob ImportJob
+	path := api.url + "/jobs/" + importJobID
+
+	jsonBody, httpCode, err := api.getJson(ctx, path, 0, nil)
+	if httpCode == http.StatusNotFound {
+		return importJob, false, nil
+	}
+	logData := log.Data{
+		"path":        path,
+		"importJobID": importJobID,
+		"httpCode":    httpCode,
+		"jsonBody":    string(jsonBody),
+	}
+	var isFatal bool
+	if err == nil && httpCode != http.StatusOK {
+		if httpCode < http.StatusInternalServerError {
+			isFatal = true
+		}
+		err = errors.New("Bad response while getting import job")
+	} else {
+		isFatal = true
+	}
+	if err != nil {
+		log.ErrorC("GetImportJob", err, logData)
+		return importJob, isFatal, err
+	}
+
+	if err := json.Unmarshal(jsonBody, &importJob); err != nil {
+		log.ErrorC("GetImportJob unmarshal", err, logData)
+		return ImportJob{}, true, err
+	}
+
+	return importJob, false, nil
+}
+
+// UpdateImportJobState tells the Import API that the state has changed of an Import job
+func (api *Client) UpdateImportJobState(ctx context.Context, jobID string, newState string) error {
+	path := api.url + "/jobs/" + jobID
+	jsonUpload := []byte(`{"state":"` + newState + `"}`)
+
+	jsonResult, httpCode, err := api.putJson(ctx, path, 0, jsonUpload)
+	logData := log.Data{
+		"path":        path,
+		"importJobID": jobID,
+		"jsonUpload":  jsonUpload,
+		"httpCode":    httpCode,
+		"jsonResult":  jsonResult,
+	}
+	if err == nil && httpCode != http.StatusOK {
+		err = errors.New("Bad HTTP response")
+	}
+	if err != nil {
+		log.ErrorC("UpdateImportJobState", err, logData)
+		return err
+	}
+	return nil
+}
+
+func (api *Client) getJson(ctx context.Context, path string, attempts int, vars url.Values) ([]byte, int, error) {
+	return callJsonAPI(ctx, api.client, "GET", path, vars)
+}
+
+func (api *Client) putJson(ctx context.Context, path string, attempts int, payload []byte) ([]byte, int, error) {
+	return callJsonAPI(ctx, api.client, "PUT", path, payload)
+}
+
+// func callJsonAPI(ctx context.Context, client *rchttp.Client, method, path string, payload interface{}) ([]byte, int, error) {
```

Comment:
do we need both/either commented strings here? If they're for future maybe add a TODO

Reply:
oops, that's my bad, another good spot